Mathlib/Order/JordanHolder.lean

```diff
@@ -230,16 +230,7 @@ theorem toList_injective : Function.Injective (@CompositionSeries.toList X _ _)
     -- Porting note: `List.nthLe_ofFn` has been deprecated but `List.get_ofFn` has a
     --               different type, so we do golf here.
     congr_fun <| List.ofFn_injective <| h.trans <| List.ofFn_congr (congr_arg Nat.succ h₁).symm _
-  cases s₁
-  cases s₂
-  -- Porting note: `dsimp at *` doesn't work. Why?
-  dsimp at h h₁ h₂
-  subst h₁
-  -- Porting note: `[heq_iff_eq, eq_self_iff_true, true_and_iff]`
-  --             → `[mk.injEq, heq_eq_eq, true_and]`
-  simp only [mk.injEq, heq_eq_eq, true_and]
-  simp only [Fin.cast_refl] at h₂
-  exact funext h₂
+  exact ext_fun h₁ h₂
```

Comment:
I would suggest instead starting the proof with `apply ext_fun` or `refine ext_fun ?_ ?_`, and solve the two subgoals directly, rather than using `have`. Either way is fine, though.

Reply:
Done, to the extent possible, in c27ed5686194986cdb15b4529225cc6fa54c7f23.
`h1` still needs to be named, because the proof of `h2` refers to it.